### PR TITLE
Added NZBIndex, NZBFriends and NZBto Providers.

### PR DESCRIPTION
**Pull request by neoatomic**
NZBIndex, NZBFriends seem to work, but still testing.
NZBto i cant test without account.

### CODE REVIEW

**Comment by miigotu** [comment]
Ahhhhhhh. Making more providers that I have to fixxxxxx.

From first look, you have to make an add_string in _get_episode_search_strings that works, and then use add_string='PROPER|REPACK' to the call to it, can use the returned strings for doSearch in proper finder. Also, check for 'REPACK' or 'PROPER' in the search results in properFinder.

Problem is that the nzb indexers are returning non-propers/repack even when searching specifically for proper/repack.
See my latest changes to newznabpy, it should be mostly copy-paste, and we want all of them to be similar.

**Comment by neoatomic** [comment]
Well i hoped it just worked. Also should be a good way for me to get some Python experience. :)  
Had it running for the night but seems to have some issues. 
The URL doesn't contain a search-string and pulls all results from the rss feed. Not sure why, as there is an option to use a search-string/show name.

```
2015-06-21 11:51:27 DEBUG    SEARCHQUEUE-DAILY-SEARCH :: [Nzbfriends] :: Nzbfriends cache update URL: http://nzbfriends.com/rss/?
2015-06-21 11:51:27 DEBUG    SEARCHQUEUE-DAILY-SEARCH :: [Nzbindex] :: Nzbindex cache update URL: https://nzbindex.com/rss/?minsize=100&max=100&complete=1
```

Also when enabling backseach it give me the below error. So backlog is probably not supported. but would be nice to suppress this error in case a user enables it.

```
2015-06-20 23:55:01 ERROR    SEARCHQUEUE-BACKLOG-259203 :: [Nzbfriends] :: Error while searching Nzbfriends, skipping: _doSearch() got an unexpected keyword argument 'epObj'
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/search.py", line 461, in searchProviders
    searchResults = curProvider.findSearchResults(show, episodes, search_mode, manualSearch, downCurQuality)
  File "/volume1/@appstore/********/var/SickRage/sickbeard/providers/generic.py", line 311, in findSearchResults
    itemList += self._doSearch(curString, 'eponly', len(episodes), epObj=epObj)
TypeError: _doSearch() got an unexpected keyword argument 'epObj'
2015-06-20 23:55:01 DEBUG    SEARCHQUEUE-BACKLOG-259203 :: [Nzbfriends] :: Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/search.py", line 461, in searchProviders
    searchResults = curProvider.findSearchResults(show, episodes, search_mode, manualSearch, downCurQuality)
  File "/volume1/@appstore/********/var/SickRage/sickbeard/providers/generic.py", line 311, in findSearchResults
    itemList += self._doSearch(curString, 'eponly', len(episodes), epObj=epObj)
TypeError: _doSearch() got an unexpected keyword argument 'epObj'
```
Anyway will try to play around with it some more.

**Comment by miigotu** [comment]
epObj is so you can us that show's exceptions and whatnot to create the right search, and to verify results are relevant. Just add epObj=None to the _doSearch definition in those and it wont give that error.

The code in findPropers should be almost exactly like it is in newznab.py now, and the _get_episode_search strings needs custom fitted from that too.

**Comment by neoatomic** [comment]
Where able to fix the epObj error and add the proper-search part. 
But when adding the _get_episode_search i get the below error.

```
2015-06-21 19:34:55 ERROR    SEARCHQUEUE-BACKLOG-294781 :: [Nzbfriends] :: Error while searching Nzbfriends, skipping: 'dict' object has no attribute 'encode'
Traceback (most recent call last):
  File "/volume1/@appstore/********/var/SickRage/sickbeard/search.py", line 461, in searchProviders
    searchResults = curProvider.findSearchResults(show, episodes, search_mode, manualSearch, downCurQuality)
  File "/volume1/@appstore/********/var/SickRage/sickbeard/providers/generic.py", line 311, in findSearchResults
    itemList += self._doSearch(curString, 'eponly', len(episodes), epObj=epObj)
  File "/volume1/@appstore/********/var/SickRage/sickbeard/providers/nzbfriends.py", line 149, in _doSearch
    "q": search_string.encode('utf-8')
AttributeError: 'dict' object has no attribute 'encode'
```

Back-search seems to work : 

```
2015-06-21 18:52:06 DEBUG    SEARCHQUEUE-BACKLOG-294781 :: [Nzbfriends] :: Search url: http://nzbfriends.com/rss/?q=The.Making.of.The.Mob.New.York. 
```

File: sickbeard/providers/nzbfriends.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            url = url.replace('&amp;', '&')
+            url = url.replace('/release/', '/download/')
+
+        return title, url
+
+    def _doSearch(self, search_string, search_mode='eponly', epcount=0, age=0, epObj=None):
+        #http://nzbindex.com/rss/?q=German&sort=agedesc&minsize=100&complete=1&max=50&more=1
+
+        params = {
+            "q": search_string.encode('utf-8')
```
Comment (miigotu) [inline sickbeard/providers/nzbfriends.py]: just remove the .encode('utf-8'), urlencode will handle it.

**Pull request closed without merging**